docs/guides/c-extension-for-python.mdx

```diff
@@ -0,0 +1,226 @@
+---
+title: Extend Qiskit in Python with C
+description: How to build a C extension for your Qiskit workflow in Python
+---
+
+# Extend Qiskit in Python with C
+
+To accelerate your Qiskit Python programs with C, you can use the Qiskit C extension for Python.
+This requires additional steps to the standalone C usage; for more details, see the [Install the Qiskit C API guide](/guides/install-c-api).
+
+<Admonition type="note">
+  These instructions have only been tested on UNIX-based systems. Windows instructions are in progress._
+</Admonition>
+
```

Comment:
I think we need to include huge warnings here that building a C extension against Qiskit is not stable for release in an external Python package, because we haven't made any guarantees about the stability of the C interface.